\documentclass[amsmath]{lni}

\usepackage{xcolor}
\usepackage[ruled]{algorithm2e}
% %\usepackage{algorithmic}
%\usepackage{amsmath}
% \usepackage{amsfonts}
\let\openbox\undefined
\usepackage{amsthm}
% \usepackage{graphicx}
\newtheorem{theorem}{Theorem}
% \newtheorem{cxampl}[theorem]{Counterexample}
\newtheorem{lemma}[theorem]{Lemma}

\newcommand{\TODO}[1]{\begingroup\color{red}#1\endgroup}
\newcommand{\NR}[1]{\begingroup\color{orange}#1\endgroup}
\newcommand{\PFS}[1]{\begingroup\color{green}#1\endgroup}

\begin{document}


\title[Phylogenetics with Mixing Hybrids]{Reconstruction of Language
  Phylogenies with ``Mixing''}

\author[Nancy Retzlaff \and Peter F.\ Stadler] {Nancy
  Retzlaff\footnote{Universität, Abteilung, Straße, Postleitzahl Ort, Land
    \email{emailaddress@author1}}
  \and
%
  Peter F.\ Stadler\footnote{Dept.\ of Computer Science and
    Interdisciplinary Center for Bioinformatics, Leipzig University,
    H{\"a}rtelstra{\ss}e 16-18, 04109 Leipzig, Germany; Max Planck
    Institute for Mathematics in the Sciences, Inselstra{\ss}e 22, D-04103
    Leipzig, Germany; Dept.\ of Theoretical Chemistry, University of
    Vienna, W{\"a}hringerstra{\ss}e 17, A-1090 Wien, Austria; Facultad de
    Ciencias, Universidad National de Colombia, Bogot{\'a}, Colombia; Santa
    Fe Institute, 1399 Hyde Park Road, Santa Fe NM 87501, USA
    \email{studla@tbi.univie.ac.at}}
}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\startpage{1} % Beginn der Seitenzählung für diesen Beitrag / Start page
\editor{Herausgeber et al.} % Names of Editors
\booktitle{Methoden und Anwendungen der Computational Humanities} % Name of book title
\year{2020}
%%% \lnidoi{18.18420/provided-by-editor-02} % if known
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\maketitle


\begin{abstract}
  super-cool abstract
\end{abstract}

\begin{keywords}
Schlagwort1 \and Schlagwort2 %Keyword1 \and Keyword2
\end{keywords}

\section{Introduction}

Phylogenetic approaches, that is, the reconstruction of historical
relationships from present-day observations has a long history in both
linguistics and biology. Most commonly, a tree models is assumed in which
lineages separate and then evolve essentially independently of each other.
While this is a very good approximation in most situations in both
biological and language evolution (borrowing nonwithstanding) there are
clear exceptions that cannot be captured by the tree model. In biology,
hybridization -- the production of offspring by mating two parents from
different varieties or species -- is not in particular in plants. In
languages, creoles or the English vocabulary serve as example. Albeit a
Germanic language, the English lexicon contains 29\% Latin and 29\% French
words, while only 25\% trace back to its Germanic origin
\cite{Finkenstaedt:73}.

As time proceeds, evolutionary events such as mutations of DNA nucleotides
and lineage-specific loss of genes or cognate words slowly increses the
dissimilarity of the evolving entities. In the tree model, so-called
additive distance measures $d(x,y)$ keep track of the number of events that
separate species or languages $x$ and $y$ from each other.  It is well
known that a unique tree can reconstructed from additive distance data
\cite{SimoesPereira:69}. Real-life data, however, are rarely if ever
additive due to unavoidable noise and measurement errors, although they are
often ``tree-like'' enough to allow the inference of a tree that is a least
a very good approximation to the true evolutionary history
\cite{Atteson:99}.

In the presence of the hybridization/mixing events, however, methods that
reconstruct trees must fail. In \cite{Prohaska:18a}, we introduced a simple
extension of the tree model describing the evolution of gene clusters by a
process known as non-homologous crossover, which leads to new genes whose
sequences consist of parts inherited from two ancestors that are adjacent
in the genome. The same model for the evolution of distances also applies
to mixing events in general by just relaxing the constraint that the
ancestors are adjacent on a linear genome, see Section~\ref{sect:R}. In
\cite{Prohaska:18a} we also devised a recognition algorithm that implicitly
reconstructs a sequence of divergence and mixing events that explains a
given distance matrix. Here we generalize this idea and provide an
approximation algorithm that is capable of inferring a sequence of events
also from imperfect data.

\section{Modelling Distances in the Presence of Mixing} 
\label{sect:R}

We consider evolution as an iterative process. Instead of speaking of genes
or languages specifically, we follow the convention of the phylogenetics
literature use the neutral term \emph{taxon}. Each taxon is associated with
data, such as the genomic DNA sequence of representative organism, a
collection of grammatics features, or a word list. With time these data
slowly change independently of each other, thus increasing the distance
$d(x,y)$ between two taxa by increments $\delta_x$ and $\delta_y$ that are
determined by each taxons individual rate of change. Once in a while, a
taxon is subdivided into two separate lineages that henceforth will evolve
independently of each other. This amounts to introducing a new taxon $z$ as
an ``offspring'' of $x$, which initially is virtually indistinguishable
from $x$. This is $d(x,z)=(x,y)$ for all taxa $y\ne x,z$ and $d(z,x)=0$.
After this ``speciation'' event, each taxon evolves independently, thus
incrementing the distances. It is not difficult to prove that this process
generates an additive distance matrix, and all additive distances matrices
can be generated in the manner \cite{Prohaska:18a}. Therefore, this simple
process amounts to the tree model, expressed in terms of the observable
distances between taxa.

A mixing or hybridization event can also be modeled in this setting
\cite{Prohaska:18a}. A hybrid $z$ is a mixture of two parents $x$ and $y$,
with $x$ contributing a fraction $\alpha$ and $y$ contributing the
remaining fraction $(1-\alpha)$. Consider the distances immediately after
the mixing event, i.e., before the lineages have time to acquire additional
changes. Then $d(x,z)=\alpha 0 + (1-\alpha) d(x,y)$ and
$d(y,z) = \alpha d(x,y) + (1-\alpha 0)$, since only the part of $z$
deriving from $y$ contributes to the distance $d(x,z)$ and \textit{vice
  versa}. The distance of any other taxon $u$ to the hybrid $z$ is is the
weighted average $d(u,z)=\alpha d(u,x)+(1-\alpha) d(u,y)$ of $u$ to its
parents $x$ and $y$. Note that for $\alpha=0$, $z$ is an identical
offspring of $y$, while for $\alpha=1$ it is an identical offspring of
$x$. The tree model outline in the previous paragraph thus is a special
case.

More formally we describe each step as follows: pick a pair of parents
$x,y\in V$ and a mixing ration $\alpha$, and pick distance
increments $\delta_u$ for all $u\in V\cup\{z\}$. Then add $z$ to $V$
and compute updated distances $d'$ as follows: 
\begin{equation} 
\begin{split} 
  d_{zx}' & = (1-\alpha)  d_{xy} +\delta_x + \delta_z \\
  d_{zy}' & =   \alpha    d_{xy} +\delta_y + \delta_z  \\
  d_{zu}' & =   \alpha    d_{xu} + (1-\alpha)  d_{yu} + \delta_u + \delta_z
  \quad\textrm{for all}  u\in V\setminus\{ x,y,z \}\\
  d_{uv}' & = d_{uv}+\delta_u+\delta_v
  \quad\textrm{for all}  u,v\in V\setminus\{ x,y,z \}
\end{split}
\label{eq:forward}
\end{equation}
it is not difficult to show that $d'$ is a metric distance measure whenever
$d$ was a metric distance. We say that a distance is an \emph{R-metric} if
it can be constructed step-by-by as outlined above. In particular, as we
have seen, every additive metric is an R-metric, i.e., R-metric are a
proper generalization of tree distances. 

Our goal is now to find a way to approximate a given distance as an
R-metric, and more precisely to give a sequence of steps of form
$(z\leftarrow x)$ and $(z\leftarrow x,y)$ in which offsprings where
produced by splitting from a single parent or by the mixing of two
parents. First we note that this cannot be done in an unambiguous manner
for less than five taxa if mixing steps may be involved. Numerical data
strongly suggest, however, that the reconstruction of the relative sequence
of events becomes unambigous beyond this unresolvable ancestral ``core''
\cite{Prohaska:18a}. 

Algorithm 2 of \cite{Prohaska:18a} in essence reverts the recursive
construction outlined above. In each step, it tries to identify a pair
$(x,z)$ corresponding to a tree-like split of lineages, or a triple
$(x,y,z)$ corresponding to a merge event. In the latter case, the mixing
ratio $\alpha$ needs to be computed. In the general case, the same result
must be obtained independently for each combination of two ``outgroups''
$u$ and $v$ distinct from the offsprin $z$ and the parents $x$ and
$y$. These consistency constraints suggest that it should be possible to
identify the correct mixing events. Simulations \cite{Prohaska:18a} suggest
that the correct series of events can be inferred most of the time. It is
not known, however, to what extent the reconstruction of the events is
truely unique. A rigorous mathematical analysis of this issue is ongoing
research.

\section{From Recognition to Approximation} 

The recognition algorithm of \cite{Prohaska:18a} pertains to exact
data. Hence it is sensitive to the unavoidable noise in real-life data.  In
order to obtain an approximate explanation of a distance matrix as a
sequence of branching and mixing steps, we have to find, in each step, the
event that fits best the data. That is, given $\textbf{D}$ we need to find
either a pair of taxa $x$ and $z$ that form a ``cherry'', i.e., branching
event such that no further event separates $x$ or $y$ from their last
common ancester, or a triple $(x,y:z)$ designating a mixing event without
further events on any of the three branches. 

For a cherry $(x:z)$ we have $\alpha=1$ in equ.(\ref{eq:forward}). This
allows us to compute the distances $\delta_x$ and $\delta_z$ between $x$
and $y$ and their last common ancestor. We obtain
$d'_{xz}=\delta_x+\delta_z$ and $d'_{ux}-d'_{uz}=\delta_x-\delta_z$ and
thus $\delta_x=(d'_{xz}+d'_{ux}-d'_{uz})/2$ and
$\delta_z=(d'_{xz}+d'_{uz}-d'_{ux})/2$, which must hold for all $u\ne x,z$.
With the auxiliary quantities $r_y=\frac{1}{n-2}\sum_{u\in V} d'_{yu}$ we
obtain $\delta_x=(d'_{xz}+r_x-r_z)/2$ and
$\delta_z=(d'_{xz}+r_z-r_x)/2$. Now, in order for $(x:z)$ to be a cherry,
$\delta_x$ must be the shortest choice over all possible $z$, and
$\delta_z$ must be the shortest choice over all possible $z$. That is,
$d'_{xz}+r_x-r_z$ -- and thus also $d'_{xz}-r_x-r_z$ -- must be minimal
over all choices of $z$ for fixed $x$, and $d'_{xz}+r_z-r_x$ -- and thus
also $d'_{xz}-r_z-r_x$ must be minimal over all choices of $x$ for fixed
$z$. Therefore, the cherry of choice $(x:z)$ is the one that minimizes the
auxiliary quantity $q_{xz}:= d'_{xz}-r_z-r_x$. Note that these are exactly
the same rules for the choice of the cherry and the computation of the
branch lengths $\delta_x$ and $\delta_z$ as in the famous Neighbor-Joining
(NJ) algorithm \cite{Saitou:87}. In contrast to NJ, however, we pick the
cherry $(x:z)$ only if the values of $S_{xz}:=d'_{xu}-d'_{zu}$ are
sufficiently similar for all $u\ne x,z$, i.e., if the variance of these
terms is small enough. Let us defer the decision of what is small enough
for later.

If we cannot pick a cherry, we have to reverse a mixing step $(x,y:z)$. The
key observation, equ.(9) in \cite{Prohaska:18a}, is that in this case the
mixing ratio can be computed as
\begin{equation}
  \alpha =
  \frac{(d'_{zu}+d'_{yv})-(d'_{zv}+d'_{yu})}%
       {(d'_{xu}+d'_{yv})-(d'_{xv}+d'_{yu})}.
  \label{eq:alpha}
\end{equation} 
This equation must hold for all pairs of outgroups $u,v\ne x,y,z$. Once
$\alpha$ is computed one can compute 
\begin{equation}
  \begin{split}
    \delta_z & = (d'_{xz}+d'_{yz}-d'_{xy})/2  \\
    \delta_x & = (d'_{xz}-(1-\alpha)d'_{xy} + S_{xyz})/(2\alpha) \\
    \delta_y & = (d'_{yz}-\alpha d'_{xy} +    S_{xyz})/(2(1-\alpha)) \\
  \end{split}
\end{equation}
where $S_{xyz}:=\alpha d'_{xu}+(1-\alpha)d'_{yu}-d'_{zu}$ must be the same
for all $u\ne x,y,z$. Of course, we estimate $S_{xyz}$ as the average over
all $u$ and use the variance of $S_{xyz}$ as a quality measure.

In order to identify the equivalent of a cherry, we need to identify a
triple $(x,y:z)$ such $\delta_z$ is minimal over all choices of parents
$x,y$, $\delta_x$ is minimal over all choices of $y,z$, and $\delta_y$ is
minimal over all choices of $x,z$. In addition, there none of the potential
cherries $(x:y)$, $(x:z)$, and $(y:z)$ are allowed to produce shorter
branches $\delta_z$, $\delta_y$ and $\delta_z$. 

\TODO{Is there a nice efficient way to find the outermost merge similar
  to the auxiliary in neighborjoining ?}

In order to measure the quality of the fit to the data, we may compare the
variance of $S_{xz} = d'_{xu}-d'_{zu}$ for the prospective cherry and the
$S_{xyx} = \alpha d'_{xu}+(1-\alpha)d'_{yu}-d'_{zu}$ for the prospective
merge, provided the estimate $\alpha$ is sufficiently far away from $0$ or
$1$.










\TODO{Probably we should directly compare how well an outer-most mixing
  event $(x,y:z)$ compares to two branching events $(x:(y:x))$}

\TODO{Pragmatische Idee: either $(x:z)$ is a true cherry or there is
  a $y$ such that that $(x,y:z)$ is mixing event. Thus we only need to
  find a $y$ that fits the data better than the cherry.}  
  



\section{Showcase Application}

\TODO{Nancy}

\section{Concluding Remarks}


\bibliography{hybrid}   
\end{document}

